elearning-frontend/src/components/admin/CreateWords.js

```diff
@@ -0,0 +1,344 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import apiClient from '../../services/api'
+import Toastify from '../../core/Toastify'
+import saveWord from '../actions/saveWord'
+import InputField from '../../core/InputField'
+import Divider from '../../core/Divider'
+import checkAdmin from '../actions/checkAdmin'
+
+const CreateWords = (props) => {
+    const navigate = useNavigate()
+
+    // Cateogry List
+    const [categoryList, setCategoryList] = useState([])
+    const [isCategoryLoading, setIsCategoryLoading] = useState(true)
+    
+    // Word
+    const [word, setWord] = useState({
+        category_id: '',
+        word: '',
+        choices: {
+            firstChoice: {
+                choice: '',
+                is_correct: false
+            },
+            secondChoice: {
+                choice: '',
+                is_correct: false
+            },
+            thirdChoice: {
+                choice: '',
+                is_correct: false
+            },
+            fourthChoice: {
+                choice: '',
+                is_correct: false
+            },
+        },
+        // correct_answer: ''
+    })
+
+    // Fetch all categories
+    const GetCategoryList = useCallback(async () => {
+        apiClient({
+            method: "get",
+            url: "/api/admin/categories"
+        }).then(response => {
+            setCategoryList(response.data.categories)
+            setIsCategoryLoading(false)
+        }).catch(error => {
+            Toastify(!((typeof error.response.data.errors) === 'undefined') ? Object.values(error.response.data.errors)[0][0]  : error.message)
+        })
+    }, [])
+
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            if(checkAdmin()) {
+                GetCategoryList()
+            }
+            else {
+                navigate('/dashboard')
+            }
+        }
+    }, [props.session])
```

Comment:
Separate the admin checker, you may place it at app.js/index.js. So you dont repeat again and again on every page.
Remember to apply DRY rule.